web-ui/src/streaming/inspection/InspectionTable.tsx

```diff
@@ -0,0 +1,305 @@
+        // active (the browser and backend only keeps a limited number of active
+        // requests and we don't want to exhaust that limit)
+        controller.abort()
+      }
+    }
+  }, [pipeline, name, relation, tableUpdater, throwError, neighborhood])
+
+  // Load quantiles for the relation we're displaying.
+  const quantileLoader = useQuantiles()
+  // Request from backend to monitor the changes for the current rows
+  useEffect(() => {
+    // The `rows.length > 0` makes sure we only send the quantile request, after
+    // the neighborhood query is established. If we do it simultaneously,
+    // somehow this led to an issue with sometimes not loading/displaying
+    // anything at all (seems like a backend bug).
+    if (relation !== undefined && rows.length > 0 && quantiles === undefined) {
+      const controller = new AbortController()
+      const url = new URL(
+        OpenAPI.BASE +
+          '/pipelines/' +
+          pipeline.descriptor.pipeline_id +
+          '/egress/' +
+          name +
+          '?format=csv&query=quantiles&mode=snapshot'
+      )
+      quantileLoader(url, setQuantiles, relation, controller).then(
+        () => {
+          // nothing to do here, the tableUpdater will update the table
+        },
+        (error: any) => {
+          if (error.name != 'AbortError') {
+            throwError(error)
+          } else {
+            // The AbortError is expected when we leave the view
+            // (controller.abort() below will trigger it)
+            // -- so nothing to do here
+          }
+        }
+      )
+
+      return () => {
+        // If we leave the view, we abort the fetch request otherwise it remains
+        // active (the browser and backend only keeps a limited number of active
+        // requests and we don't want to exhaust that limit)
+        controller.abort()
+      }
+    }
+  }, [pipeline, name, relation, quantileLoader, rows, throwError, quantiles])
+
+  // If the user presses the back or next button, we update the neighborhood
+  // which will trigger update the rows in the table.
+  const handlePaginationModelChange = useCallback(
+    (newPaginationModel: GridPaginationModel) => {
+      if (!relation) {
+        return
+      }
+      if (paginationModel.page > newPaginationModel.page) {
+        // Going backward, we can only go backward as long as we have at least
+        // one item with a genId < 0
+        const minRow = rows.find((row: any) => row.genId < 0)
+        if (!minRow) {
+          pushMessage({
+            message: `You reached the beginning of the ${relation.name} relation.`,
+            key: new Date().getTime(),
+            color: 'info'
+          })
+          return
+        }
+        setRows([])
+        setNeighborhood({
+          ...DEFAULT_NEIGHBORHOOD,
+          anchor: objectToAnchor(relation, minRow)
+        })
+      } else {
+        if (rows[rows.length - 1].genId === PAGE_SIZE) {
```

Comment:
Right now if weight>1 we display it as a single row (but  the weight can be viewed as a column of the table). This might be confusing for users not familiar with how z-sets/weights work.

Alternatives:
- It would be a bit painful (but not impossible I guess) to build a UI table where it handles the weights>1 by displaying the row multiple times. 
- Another alternative would be to just send the same rows multiple times (e.g., ensure weight=1 for all rows) from the server.

Reply:
make the weight column a different style